protected/humhub/helpers/DataTypeHelper.php

```diff
@@ -0,0 +1,628 @@
+    {
+        $param = $throw ? '$value' : false;
+
+        if ($strict) {
+            $types = 'integer';
+            $input = $value;
+        } else {
+            $types = ['integer', null];
+            $input = filter_var($value, FILTER_VALIDATE_INT, FILTER_NULL_ON_FAILURE);
+        }
+
+        return self::checkType($input, $types, false, $param) === null ? null : $input;
+    }
+
+    /**
+     * @param mixed $value value to be tested or converted
+     * @param bool $strict indicates if strict comparison should be performed:
+     * ``
+     * - if True, `$value` must already be a scalar value.
+     * - if False, `NULL` is also allowed (not throwing an exception, if `$throw` is True).
+     * ``
+     *
+     * @since 1.16
+     */
+    public static function filterScalar($value, bool $strict = null, bool $throw = false)
+    {
+        $param = $throw ? '$value' : false;
+        $types = $strict ? ['is_scalar'] : [null, 'is_scalar'];
+
+        return self::checkType($value, $types, false, $param) === null ? null : $value;
+    }
+
+    /**
+     * @param mixed $value value to be tested or converted
+     * @param bool $strict indicates if strict comparison should be performed:
+     * ``
+     * - if True, `$value` must already be of type `string`.
+     * - if False, a conversion to `string` is attempted.
+     * ``
+     *
+     * @since 1.16
+     */
+    public static function filterString($value, bool $strict = false, bool $throw = false): ?string
+    {
+        $param = $throw ? '$value' : false;
+        $types = $strict ? 'string' : ['string', null, Stringable::class, 'is_scalar'];
+
+        switch (self::checkType($value, $types, false, $param)) {
+            case 'string':
+                return $value;
+            case 'NULL':
+                return null;
+            case Stringable::class:
+                return $value->__toString();
+            case 'is_scalar':
+                return (string)$value;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param mixed $value value to be tested or converted
+     * @param bool $strict indicates if strict comparison should be performed:
+     * ``
+     * - if True, `$value` must already be of type `int`.
+     * - if False, a conversion to `int` is attempted.
+     * ``
+     *
+     * @since 1.16
+     */
+    public static function checkInt($value, bool $strict = false): ?int
+    {
+        // check if strict
+        if (($strict && !is_int($value)) || !is_scalar($value)) {
+            return null;
+        }
+
+        return filter_var($value, FILTER_VALIDATE_INT, FILTER_NULL_ON_FAILURE);
+    }
+
+    /**
+     * @param string|object $class
+     * @param bool $autoload
+     *
+     * @return array|null
+     * @see https://www.php.net/manual/en/function.class-uses.php#122427
+     */
+    public static function &classUsesTraits($class, bool $autoload = true): ?array
```

Comment:
Do we really need to return the array reference here?

Reply:
As we currently never return references like this, I would prefer not to start with it now.
I also hope that PHP internally only copies the array when it is really necessary.

Reply:
> As we currently never return references like this, I would prefer not to start with it now.

Ok.

> I also hope that PHP internally only copies the array when it is really necessary.

Unfortunately, that's not the case. The [PHP Manual](https://www.php.net/manual/en/language.types.array.php#example-208) states under Example 17:

> Array assignment always involves value copying. Use the [reference operator](https://www.php.net/manual/en/language.operators.php) to copy an array by reference.

Reply:
@martin-rueegg As I understand it, the example refers more to changes of arrays. 

Please see: https://www.php.net/manual/en/language.references.return.php

Reply:
Ok, that's good news for me. Thanks for the reference.